crates/sparrow-runtime/src/metadata/raw_metadata.rs

```diff
@@ -106,75 +105,57 @@ impl RawMetadata {
         })
     }
 
-    /// Create a `RawMetadata` from a parquet string path and object store registry
+    /// Create a `RawMetadata` from a parquet string path and object store registry.
+    ///
+    /// This uses `object_store` to asynchronously retrieve only the Parquet
+    /// footer to determine the schema.
     async fn try_from_parquet(
         path: &str,
-        object_store_registry: &ObjectStoreRegistry,
+        object_stores: &ObjectStoreRegistry,
     ) -> error_stack::Result<Self, Error> {
-        let object_store_url = ObjectStoreUrl::from_str(path)
+        let url = ObjectStoreUrl::from_str(path)
             .change_context_lazy(|| Error::ObjectStore(path.to_owned()))?;
-        let object_store_key = object_store_url
-            .key()
+
+        let parquet_file = ParquetFile::try_new(object_stores, url)
+            .await
             .change_context_lazy(|| Error::ObjectStore(path.to_owned()))?;
-        match object_store_key {
-            ObjectStoreKey::Local => {
-                let path = object_store_url
-                    .path()
-                    .change_context_lazy(|| Error::ObjectStore(path.to_owned()))?
-                    .to_string();
-                // The local paths are formatted file:///absolute/path/to/file.file
-                // The Object Store path strips the prefix file:/// but we need to add the
-                // root slash back prior to opening the file.
-                let path = format!("/{}", path);
-                let path = std::path::Path::new(&path);
-                Self::try_from_parquet_path(path)
-            }
-            _ => {
-                let download_file = NamedTempFile::new().map_err(|_| Error::Download)?;
-                object_store_url
-                    .download(object_store_registry, download_file.path())
-                    .await
-                    .change_context_lazy(|| Error::Download)?;
-                Self::try_from_parquet_path(download_file.path())
-            }
-        }
+
+        Self::from_raw_schema(parquet_file.schema)
     }
 
-    /// Create a `RawMetadata` from a CSV string path and object store registry
+    /// Create a `RawMetadata` from a CSV string path and object store registry.
+    ///
+    /// For CSV, this currently needs to download a local copy of the file, since
+    /// Arrow does not (as of 2023-July-06) support reading CSV using `AsyncWrite`.
```

Comment:
Is there a feature request/issue number we could follow to see if they'll be supporting this in the near future?

Reply:
Broke the CSV case out into https://github.com/kaskada-ai/kaskada/issues/486